### PR TITLE
[BugFix](hdfsScanner) Fix that column nums might continuously expand if the the chunk filtered is empty every time

### PR DESCRIPTION
**Pull request by caiconghui**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [x] 2.5

### CODE REVIEW

File: be/src/exec/hdfs_scanner.cpp [hunk 1]
```diff
@@ -468,6 +468,9 @@ Status HdfsScannerContext::append_or_update_not_existed_columns_to_chunk(ChunkPt
         ck->append_or_update_column(std::move(col), slot_desc->id());
     } else {
         for (auto* slot_desc : not_existed_slots) {
+            if (ck->is_slot_exist(slot_desc->id())) {
+                continue;
+            }
```
Comment (imay) [inline be/src/exec/hdfs_scanner.cpp:473]: Does it hurt performance if there are thousands of columns in the chunk?

**Review by imay — COMMENTED** [review]
Can you comment at what situation, this bug will be hit? And also is it possible to add some test to cover this fix>

File: be/src/exec/hdfs_scanner.cpp (hunk 1 above)
Comment (caiconghui) [reply be/src/exec/hdfs_scanner.cpp:473]: I think it is ok now, thousands of columns in the chunk is rare and, the column num append to chunk is small, the data structure in chunk for detect id whether exist is  map

**Comment by caiconghui** [comment]
> Can you comment at what situation, this bug will be hit? And also is it possible to add some test to cover this fix>

Status HiveDataSource::get_next(RuntimeState* state, ChunkPtr* chunk) {
    if (_no_data) {
        return Status::EndOfFile("no data");
    }

    do {
        RETURN_IF_ERROR(_init_chunk_if_needed(chunk, _runtime_state->chunk_size()));
        RETURN_IF_ERROR(_scanner->get_next(state, chunk));
    } while ((*chunk)->num_rows() == 0);

    // The column order of chunk is required to be invariable. In order to simplify the logic of each scanner,
    // we force to reorder the columns of chunk, so scanner doesn't have to care about the column order anymore.
    // The overhead of reorder is negligible because we only swap columns.
    ChunkHelper::reorder_chunk(*_tuple_desc, chunk->get());

    return Status::OK();
}

if num_rows() is zero every turn, the bug will be hit

**Comment by DorianZheng** [comment]
Please add a UT

**Comment by DorianZheng** [comment]
Discuss with @caiconghui, this issue has been fixed. I will close it and feel free to re-open it when there is update.

**Pull request closed without merging**